.github/workflows/nubots.yaml

```diff
@@ -0,0 +1,179 @@
+name: NUbots CI
+
+# Controls when the action will run.
+on:
+  # Triggers the workflow on push or pull request events but only for the main branch
+  push:
+    branches: [main]
+  pull_request:
+    branches: [main]
+
+# A workflow run is made up of one or more jobs that can run sequentially or in parallel
+jobs:
+  # Build the docker image
+  build_docker:
+    name: "Build docker image"
+
+    # The type of runner that the job will run on
+    runs-on: ubuntu-20.04
+
+    # We output the image tag that we create so the other jobs can use it
+    outputs:
+      image: ${{ steps.image_output.outputs.image }}
+
+    # Steps represent a sequence of tasks that will be executed as part of the job
+    steps:
+      # Cancels previously running jobs for this workflow
+      - name: Cancel Workflow Action
+        uses: styfle/cancel-workflow-action@0.9.1
+
+      # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
+      - name: Checkout Code
+        uses: actions/checkout@v2
+
+      # Setup docker buildx
+      - name: 🐳 Set up docker buildx 🐳
+        uses: docker/setup-buildx-action@v1
+
+      - name: Login to DockerHub
+        uses: docker/login-action@v1
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_PASSWORD }}
+
+      # Build the docker image
+      - name: 🐳 Build the docker image 🐳
+        uses: docker/build-push-action@v2
+        with:
+          pull: true
+          tags: "nubots/nubots:${{ hashFiles('docker/**') }}"
+          file: docker/Dockerfile
+          context: docker
+          build-args: |
+            platform=generic
+          push: true
+          cache-from: type=registry,ref=nubots/nubots:generic
+          cache-to: type=inline
+
+      - id: image_output
+        name: Output the created image
+        run: echo "::set-output name=image::nubots/nubots:${{ hashFiles('docker/**') }}"
+
+  # Build the codebase
+  build_nubots:
+    name: "Build and test NUbots"
+
+    # The type of runner that the job will run on
+    runs-on: ubuntu-20.04
+    needs: build_docker
+
+    # Run on the container we just built
+    container:
+      image: "${{ needs.build_docker.outputs.image }}"
+      options: --user 0:0
+
+    # Steps represent a sequence of tasks that will be executed as part of the job
+    steps:
+      # Cancels previously running jobs for this workflow
+      - name: Cancel Workflow Action
+        uses: styfle/cancel-workflow-action@0.9.1
+
+      # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
+      - name: Checkout Code
+        uses: actions/checkout@v2
+
+      - name: Configure the code
+        run: |
+          mkdir -p build
+          cd build
+          cmake .. -GNinja \
+                   -DCMAKE_BUILD_TYPE=Release \
+                   -DCMAKE_TOOLCHAIN_FILE=/usr/local/toolchain.cmake \
+                   -DBUILD_TESTS=ON \
+                   -DCI_BUILD=ON
```

Comment:
Why are we not using the b script?

Reply:
Good question. I'll try it with the `b` script.

Reply:
That worked. Just had to move the build directory :+1: